tests/simulators/carla/test_actions.py

```diff
@@ -0,0 +1,118 @@
+import os
+from pathlib import Path
+import socket
+import subprocess
+import time
+
+import pytest
+
+try:
+    import carla
+
+    from scenic.simulators.carla import CarlaSimulator
+except ModuleNotFoundError:
+    pytest.skip("carla package not installed", allow_module_level=True)
+
+from tests.utils import compileScenic, sampleScene
+
+
+def checkCarlaPath():
+    CARLA_ROOT = os.environ.get("CARLA_ROOT")
+    if not CARLA_ROOT:
+        pytest.skip("CARLA_ROOT env variable not set.")
+    return CARLA_ROOT
+
+
+def isCarlaServerRunning(host="localhost", port=2000):
+    with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as sock:
+        sock.settimeout(1)
+        try:
+            sock.connect((host, port))
+            return True
+        except (socket.timeout, ConnectionRefusedError):
+            return False
+
+
+@pytest.fixture(scope="package")
+def getCarlaSimulator():
```

Comment:
```suggestion
def getCarlaSimulator(getAssetPath):
```

Reply:
In order to use this fixture, we need to be update to `session` to avoid the following error:
```
ScopeMismatch: You tried to access the function scoped fixture getAssetPath with a package scoped request object, involved factories:
tests/simulators/carla/test_actions.py:37:  def getCarlaSimulator(getAssetPath)
tests/conftest.py:43:  def getAssetPath()
```